File: Remove the balls-gfg/Remove the balls.java
class Solution {
  public:
    int findLength(vector<int> &color, vector<int> &radius) {
        int n = color.size();
        stack<int> st;
        
        for (int i = 0; i < n; ++i) {
            if (!st.empty() && color[i] == color[st.top()] 
            && radius[i] == radius[st.top()]) {
                st.pop();
            } else {
                st.push(i);
            }
        }
        return st.size();
    }
};

File: Remove the balls-gfg/Remove the balls.cpp
class Solution {
    public int findLength(int[] color, int[] radius) {
        // code here
        int n= color.length;
        Stack<Integer> st = new Stack<>();
        for(int i=0; i<n; i++){
            if(!st.isEmpty() && color[i]==color[st.peek()] 
            && radius[i]==radius[st.peek()])
            st.pop();
            else 
            st.push(i);
        }
        return st.size();
    }
}
